crates/valence_entity/src/active_status_effects.rs

```diff
@@ -0,0 +1,222 @@
+use bevy_ecs::prelude::*;
+
+use crate::status_effects::StatusEffect;
+
+/// [`Component`] that stores the [`ActiveStatusEffect`]s of an [`Entity`].
+#[derive(Component, Default, Debug)]
+pub struct ActiveStatusEffects {
+    active_effects: Vec<ActiveStatusEffect>,
+    new_effects: Vec<ActiveStatusEffect>,
+}
+
+impl ActiveStatusEffects {
+    /// Adds a new [`ActiveStatusEffect`] to the [`ActiveStatusEffects`].
+    ///
+    /// ## Note
+    ///
+    /// It actually adds the [`ActiveStatusEffect`] to the new effects. The
+    /// [`ActiveStatusEffect`] will be added to the active effects in the next
+    /// tick (more specifically, during the upcoming [`EventLoopPostUpdate`]).
+    /// If the [`ActiveStatusEffect`] is already in the new effects, it will be
+    /// replaced.
+    ///
+    /// [`EventLoopPostUpdate`]: https://docs.rs/valence_server/0.2.0-alpha.1+mc.1.20.1/valence_server/event_loop/struct.EventLoopPostUpdate.html
+    pub fn add(&mut self, effect: ActiveStatusEffect) {
```

Comment:
why not check in `active_effects` and `new_effects` here?

Reply:
wdym?